src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -64,10 +69,80 @@ void parallel_reduce (const TeamMember& team,
 
    result = local_tmp;
   } else {
+    const ValueType initial(result);
     Kokkos::parallel_reduce(Kokkos::TeamThreadRange(team, begin, end), lambda, result);
+    result += initial;
   }
 }
 
+/*
+ * Computes a reduction over a routine described by 'input' for entries 'scalarize(input)([begin,end))'
+ * The variable 'input' should be an operator s.t. 'input(k)' returns a Pack. The computed reduction is
+ * added to the value 'result'.
+ */
+template <bool Serialize, typename TeamMember, typename InputProvider, typename ValueType>
+static KOKKOS_INLINE_FUNCTION
+void view_reduction (const TeamMember& team,
+                     const int begin,
+                     const int end,
+                     const InputProvider& input,
+                     ValueType& result)
+{
+  using PackType = typename std::remove_reference<decltype(input(0))>::type;
+  constexpr int vector_size = PackType::n;
+
+  // Perform a packed reduction over scalar indices
+  if (true) { // !Serialize) {
+    const bool has_garbage_begin = (begin%vector_size != 0 ? true : false);
```

Comment:
Style: `begin%vector_size != 0` is a boolean, so ternary with true/false as outputs is redundant. Just write `const bool has_garbage_begin = begin%vector_size != 0`.